### PR TITLE
Practical 05 code review request

### PR DESCRIPTION
**Pull request by nicholaswilliams-xyz**
Hello, @heshan8, 

Please review all my python files.

Best regards,

Nicholas.

### CODE REVIEW

File: practical_05/hex_colours.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+for colour, hex_code in COLOUR_TO_HEX.items():
+    print(f"Colour: {colour:<13}, {hex_code}")
+
+colour_name = input("Enter a colour: ").lower()
+while colour_name != "":
+    try:
+        print(colour_name, "is", COLOUR_TO_HEX[colour_name].lower())
+    except KeyError:
+        print("Invalid short state. Incorrect key used.")
```
Comment (heshan8) [inline practical_05/hex_colours.py:19]: Did you mean "Invalid colour name"?

**Review by heshan8 — COMMENTED** [review]
Nice work :)

File: practical_05/emails.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    else:
+        proper_name = input("Name: ")
+        NAME_TO_EMAIL[proper_name] = email
+
+    email = input("Email: ")
+
+left_align = len(max(NAME_TO_EMAIL, key=len))
+
+for name, email in NAME_TO_EMAIL.items():
+    print(f"{name:<{left_align + 1}}: {email}")
```
Comment (heshan8) [inline practical_05/emails.py:26]: Use functions to break up the program, so it's easier to understand 😄.

**Review by heshan8 — COMMENTED** [review]
Nice work.

File: practical_05/wimbledon.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+Actual completion time:
+"""
+
+FILENAME = 'wimbledon.csv'
+victor_to_wins = {}
+winning_countries = set()
+wimbledon_data = []
+
+
+def main():
```
Comment (heshan8) [inline practical_05/wimbledon.py:13]: Missing docstrings

File: practical_05/wimbledon.py [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+            victor_to_wins[line[2]] = 1  # Otherwise, add them to dictionary with 1 victory
+        winning_countries.add(line[1])
+
+    print("\nWimbledon Champions:")
+    left_align = len(max(victor_to_wins, key=len))
+    for name, victories in victor_to_wins.items():
+        print(f"{name:<{left_align + 1}}: {victories}")
+
+    print(f"\nThe {len(winning_countries)} countries that won Wimbledon were:")
+    print(", ".join(country for country in sorted(winning_countries)))
```
Comment (heshan8) [inline practical_05/wimbledon.py:47]: Good use of functions 👍

**Review by heshan8 — COMMENTED** [review]
Nice work.

File: practical_05/hex_colours.py (hunk 1 above)
Comment (nicholaswilliams-xyz) [reply practical_05/hex_colours.py:19]: I did indeed

**Pull request merged**